repose-aggregator/services/phone-home/src/main/scala/org/openrepose/core/services/phonehome/PhoneHomeService.scala

```diff
@@ -0,0 +1,188 @@
+import org.openrepose.core.spring.ReposeSpringProperties
+import org.openrepose.core.systemmodel.{FilterList, ServicesList, SystemModel}
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import play.api.libs.json.Json.JsValueWrapper
+import play.api.libs.json.{JsNull, Json, Writes}
+
+import scala.collection.JavaConverters._
+
+/**
+ * A service which sends Repose usage data to a data collection point. This data may be used to determine common usage
+ * patterns.
+ * <p/>
+ * Data that may be sent includes:
+ * - Filter chain
+ * - Services
+ * - Contact information
+ * <p/>
+ * Note that the lifecycle of this service is self-managed, and the configuration for this service is read directly
+ * from the system model configuration file. Additionally, usage data will be sent whenever an update to the system
+ * model is observed, as long as the phone home service is active.
+ */
+@Named
+class PhoneHomeService @Inject()(@Value(ReposeSpringProperties.CORE.REPOSE_VERSION) reposeVer: String,
+                                 @Value(ReposeSpringProperties.CORE.CONFIG_ROOT) confDir: String,
+                                 configurationService: ConfigurationService,
+                                 akkaServiceClient: AkkaServiceClient)
+  extends LazyLogging {
+
+  private final val msgLogger = LoggerFactory.getLogger("phone-home-message")
+
+  private var systemModel: SystemModel = _
+
+  @PostConstruct
+  def init(): Unit = {
+    logger.debug("Registering system model listener")
+    configurationService.subscribeTo(
+      "system-model.cfg.xml",
+      SystemModelConfigurationListener,
+      classOf[SystemModel]
+    )
+  }
+
+  private def sendUpdate(): Unit = {
+    logger.trace("sendUpdate method called")
+
+    val staticSystemModel = systemModel // Pin the system model in case an update occurs while processing
+
+    Option(staticSystemModel.getPhoneHome) match {
+      case Some(phoneHome) if phoneHome.isEnabled =>
+        logger.debug("Sending usage data update to data collection service")
+
+        sendUpdateMessage(
+          phoneHome.getCollectionUri,
+          buildUpdateMessage(phoneHome.getOriginServiceId, phoneHome.getContactEmail)
+        )
+      case Some(phoneHome) if !phoneHome.isEnabled =>
+        val updateMessage = buildUpdateMessage(phoneHome.getOriginServiceId, phoneHome.getContactEmail)
+        logger.warn(buildLogMessage(
+          "Did not attempt to send usage data on update -- the phone home service is not enabled",
+          updateMessage,
+          phoneHome.getCollectionUri
+        ))
+        msgLogger.info(updateMessage)
+      case _ =>
+        val updateMessage = buildUpdateMessage()
+        logger.warn(buildLogMessage(
+          "Did not attempt to send usage data on update -- the phone home service is not enabled",
+          updateMessage,
+          "<phone-home-collection-uri>"
```

Comment:
The error messages were basically the same, so I made a tiny function to generate them. Since in this case we don't have a collection uri, printing <phone-home-collection-uri> instead hopefully shows that it's something that will need to be templated in.